services/learn-card-network/brain-service/test/profiles.spec.ts

```diff
@@ -1410,6 +1410,46 @@ describe('Profiles', () => {
 
             jest.useRealTimers();
         });
+
+        it('allows setting the expiration date', async () => {
+            jest.useFakeTimers().setSystemTime(new Date('02-06-2023'));
+
+            await userA.clients.fullAuth.profile.generateInvite({
+                challenge: 'nice',
+                expiration: new Date('02-06-2025').getTime(),
+            });
+
+            jest.setSystemTime(new Date('02-06-2025'));
+
+            await expect(
+                userB.clients.fullAuth.profile.connectWithInvite({
+                    challenge: 'nice',
+                    profileId: 'usera',
+                })
+            ).resolves.not.toThrow();
+
+            jest.useRealTimers();
+        });  
+
+        it('the generation date cannot be updated after creation', async () => {
+            jest.useFakeTimers().setSystemTime(new Date('02-06-2023'));
+
+            await userA.clients.fullAuth.profile.generateInvite({
+                challenge: 'nice',
+                expiration: new Date('02-06-2025').getTime(),
+            });
+
+            jest.setSystemTime(new Date('02-06-2025'));
+
+            await expect(
+                userB.clients.fullAuth.profile.connectWithInvite({
+                    challenge: 'nice',
+                    profileId: 'usera',
+                })
+            ).resolves.not.toThrow();
+
+            jest.useRealTimers();
+        });
```

Comment:
I'm not sure what this test is doing D=

Reply:
okay I rewrote it with a more clear "allows connections with challenges before they expire"